src/file.cr

```diff
@@ -237,23 +240,34 @@ class File < IO::FileDescriptor
     basename(path).chomp(suffix)
   end
 
+  private alias UserRep = System::User | String | UInt32 | Int32
+  private alias GroupRep = System::Group | String | UInt32 | Int32
+
   # Changes the owner of the specified file.
   #
   # ```
-  # File.chown("/foo/bar/baz.cr", 1001, 100)
-  # File.chown("/foo/bar", gid: 100)
+  # File.chown("/foo/bar/baz.cr", "alice", "admin")
+  # File.chown("/foo/bar", group: "admin")
   # ```
   #
   # Unless *follow_symlinks* is set to `true`, then the owner symlink itself will
   # be changed, otherwise the owner of the symlink destination file will be
   # changed. For example, assuming symlinks as `foo -> bar -> baz`:
   #
   # ```
-  # File.chown("foo", gid: 100)                        # changes foo's gid
-  # File.chown("foo", gid: 100, follow_symlinks: true) # changes baz's gid
+  # File.chown("foo", group: "admin")                        # changes foo's group to "admin"
+  # File.chown("baz", group: "admin", follow_symlinks: true) # changes baz's group to "admin"
+  # ```
+  #
+  # Raises on error.
+  def self.chown(path : String, owner : UserRep? = nil, group : GroupRep? = nil, follow_symlinks : Bool = false) : Nil
+    Crystal::System::File.chown(path, System::User.raw_uid?(owner) || -1, System::Group.raw_gid?(group) || -1, follow_symlinks)
+  end
+
+  # Same as `chown()` but instead returns a `Bool` indicating success.
   # ```
-  def self.chown(path, uid : Int = -1, gid : Int = -1, follow_symlinks = false)
-    Crystal::System::File.chown(path, uid, gid, follow_symlinks)
+  def self.chown?(path : String, owner : UserRep? = nil, group : GroupRep? = nil, follow_symlinks : Bool = false) : Bool
```

Comment:
I'd rather not add new unrelated methods in this PR, like `File.chown?` and `File#owner=`, `File#group=`.

Reply:
I would argue that this is directly related. And absolutely necessary. This functionality was a major reason adding `User`s and `Group`s.